aries_cloudagent/protocols/issue_credential/v2_0/routes.py

```diff
@@ -0,0 +1,1343 @@
+from ....messaging.valid import (
+    INDY_CRED_DEF_ID,
+    INDY_DID,
+    INDY_SCHEMA_ID,
+    INDY_VERSION,
+    UUIDFour,
+    UUID4,
+)
+from ....storage.error import StorageError, StorageNotFoundError
+from ....wallet.base import BaseWallet
+from ....wallet.error import WalletError
+from ....utils.outofband import serialize_outofband
+from ....utils.tracing import trace_event, get_timer, AdminAPIMessageTracingSchema
+
+from ...problem_report.v1_0 import internal_error
+from ...problem_report.v1_0.message import ProblemReport
+
+from .manager import V20CredManager, V20CredManagerError
+from .message_types import SPEC_URI
+from .messages.cred_format import V20CredFormat
+from .messages.cred_offer import V20CredOfferSchema
+from .messages.cred_proposal import V20CredProposal
+from .messages.inner.cred_preview import V20CredPreview, V20CredPreviewSchema
+from .models.cred_ex_record import V20CredExRecord, V20CredExRecordSchema
+from .models.detail.dif import V20CredExRecordDIFSchema
+from .models.detail.indy import V20CredExRecordIndySchema
+
+
+class V20IssueCredentialModuleResponseSchema(OpenAPISchema):
+    """Response schema for v2.0 Issue Credential Module."""
+
+
+class V20CredExRecordListQueryStringSchema(OpenAPISchema):
+    """Parameters and validators for credential exchange record list query."""
+
+    connection_id = fields.UUID(
+        description="Connection identifier",
+        required=False,
+        example=UUIDFour.EXAMPLE,  # typically but not necessarily a UUID4
+    )
+    thread_id = fields.UUID(
+        description="Thread identifier",
+        required=False,
+        example=UUIDFour.EXAMPLE,  # typically but not necessarily a UUID4
+    )
+    role = fields.Str(
+        description="Role assigned in credential exchange",
+        required=False,
+        validate=validate.OneOf(
+            [
+                getattr(V20CredExRecord, m)
+                for m in vars(V20CredExRecord)
+                if m.startswith("ROLE_")
+            ]
+        ),
+    )
+    state = fields.Str(
+        description="Credential exchange state",
+        required=False,
+        validate=validate.OneOf(
+            [
+                getattr(V20CredExRecord, m)
+                for m in vars(V20CredExRecord)
+                if m.startswith("STATE_")
+            ]
+        ),
+    )
+
+
+class V20CredExRecordDetailSchema(OpenAPISchema):
+    """Credential exchange record and any per-format details."""
+
+    cred_ex_record = fields.Nested(
+        V20CredExRecordSchema,
+        required=False,
+        description="Credential exchange record",
+    )
+    indy = fields.Nested(
+        V20CredExRecordIndySchema,
+        required=False,
+    )
+    dif = fields.Nested(
+        V20CredExRecordDIFSchema,
+        required=False,
+    )
+
+
+class V20CredExRecordListResultSchema(OpenAPISchema):
+    """Result schema for credential exchange record list uery."""
```

Comment:
```suggestion
    """Result schema for credential exchange record list query."""
```

Reply:
Are you volunteering to fix all the typos in all the comments? If so, you're a champ :-D